rust/main/agents/relayer/src/relayer.rs

```diff
@@ -1315,7 +1360,39 @@ mod test {
     #[tokio::test]
     #[tracing_test::traced_test]
     async fn test_failed_build_mailboxes() {
-        let settings = generate_test_relayer_settings();
+        let temp_dir = tempfile::tempdir().unwrap();
+        let db_path = temp_dir.path();
+
+        let chains = vec![
+            (
+                KnownHyperlaneDomain::Arbitrum.to_string(),
+                generate_test_chain_conf(
+                    HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum),
+                    None,
+                    "https://sepolia-rollup.arbitrum.io/rpc",
```

Comment:
do we expect these URLs to always be live in these tests? like are we actually making requests to these URLs in these tests, and if this RPC or any others go down, do these tests fail?

If so, I'd prefer for us to not have unit tests make any requests because this feels like a recipe for flakiness